force-app/main/default/classes/OpportunityTriggerHandler.cls

```diff
@@ -0,0 +1,186 @@
+public class OpportunityTriggerHandler extends TriggerHandler {
+  public override void beforeInsert() {
+    // Set default Type for new Opportunities
+    for (Opportunity opp : (List<Opportunity>) Trigger.new) {
+      if (opp.Type == null) {
+        opp.Type = 'New Customer';
+      }
+    }
+  }
+
+  public override void beforeUpdate() {
+    // Validate that the amount is greater than 5000
+    for (Opportunity opp : (List<Opportunity>) Trigger.new) {
+      if (opp.Amount < 5000) {
+        opp.addError('Opportunity amount must be greater than 5000');
+      }
+    }
+
+    // Set the primary contact to the CEO if the field is empty
+    Set<Id> accountIds = new Set<Id>();
+    for (Opportunity opp : (List<Opportunity>) Trigger.new) {
+      if (opp.AccountId != null) {
+        accountIds.add(opp.AccountId);
+      }
+    }
+
+    if (!accountIds.isEmpty()) {
+      Map<Id, Contact> ceoContacts = new Map<Id, Contact>(
+        [
+          SELECT Id, AccountId
+          FROM Contact
+          WHERE AccountId IN :accountIds AND Title = 'CEO'
+        ]
+      );
+
+      for (Opportunity opp : (List<Opportunity>) Trigger.new) {
+        if (
+          opp.Primary_Contact__c == null &&
+          ceoContacts.containsKey(opp.AccountId)
+        ) {
+          opp.Primary_Contact__c = ceoContacts.get(opp.AccountId).Id;
+        }
```

Comment:
Does this look very similar to something else you are doing in another trigger context? Maybe you can combine that functionality together so that you only need to write that code once.